src/applications/app.helpers.js

```diff
@@ -74,3 +74,111 @@ export function isParcel(appOrParcel) {
 export function objectType(appOrParcel) {
   return isParcel(appOrParcel) ? "parcel" : "application";
 }
+
+export function validateArgumentsTypeAPI(
+  appName,
+  applicationOrLoadingFn,
+  activityFn,
+  customProps
+) {
+  if (typeof appName !== "string" || appName.length === 0)
+    throw Error(
+      formatErrorMessage(
+        20,
+        __DEV__ &&
+          `The first argument to registerApplication must be a non-empty string 'appName'`
+      )
+    );
+
+  if (!applicationOrLoadingFn)
+    throw Error(
+      formatErrorMessage(
+        23,
+        __DEV__ && "The application or loading function is required"
+      )
+    );
+
+  if (typeof activityFn !== "function")
+    throw Error(
+      formatErrorMessage(
+        24,
+        __DEV__ && `The activityFunction argument must be a function`
+      )
+    );
+
+  if (
+    !!customProps &&
+    (typeof customProps !== "object" || Array.isArray(customProps))
```

Comment:
Changed validation for `customProps` because, since it's optional, it should skip the validation if it's not present